lib/rspec/core/metadata_filter.rb

```diff
@@ -15,22 +15,19 @@ def apply?(predicate, filters, metadata)
         # @private
         def filter_applies?(key, value, metadata)
           silence_metadata_example_group_deprecations do
-            return filter_applies_to_any_value?(key, value, metadata) if Array === metadata[key] && !(Proc === value)
             return location_filter_applies?(value, metadata)          if key == :locations
             return id_filter_applies?(value, metadata)                if key == :ids
             return filters_apply?(key, value, metadata)               if Hash === value
 
             return false unless metadata.key?(key)
+            return true if value.is_a?(TrueClass) && !!metadata[key]
```

Comment:
You're right :).


Speaking of `is_a?`, we avoid that in RSpec, for the simple reason that users pass all manner of objects to us and we can't necessarily trust that it responds to `is_a?` or that its implementation does what we expect.  For example, `BasicObject` does not respond to `is_a?` or a user may have redefined `is_a?` for their domain (e.g. `user.is_a?(:admin)`).

Instead, we always favor `klass === object` over `object.is_a?(klass)`. In this case, that would be `TrueClass === value`.  If the user is redefining `===` on core built in classes than we're not going to support them (and we've never seen that happen).